analysis/priestdiscipline/src/modules/spells/Penance.js

```diff
@@ -48,19 +31,35 @@ class Penance extends Analyzer {
       return;
     }
 
-    this.eventGrouper.processEvent(event);
-
-    event.penanceBoltNumber = this.currentBoltNumber;
+    event.penanceBoltNumber = this._boltCount;
+    this._boltCount += 1;
   }
 
   onHeal(event) {
     if (!Penance.isPenance(event.ability.guid)) {
       return;
     }
 
-    this.eventGrouper.processEvent(event);
+    event.penanceBoltNumber = this._boltCount;
+    this._boltCount += 1;
+  }
+
+  onCast(event) {
+    this._casts += 1;
+    this._missedBolts += this._expectedBoltCount - this._boltCount;
```

Comment:
If you determine missed bolts of the last cast when the new cast happens then you run into the issue of never checking the last cast of the fight